subsys/bluetooth/mesh/shell/rpr.c

```diff
@@ -38,7 +38,7 @@ static void rpr_scan_report(struct bt_mesh_rpr_cli *cli,
 		uint8_t len, type;
 		uint8_t data[31];
 
-		len = net_buf_simple_pull_u8(adv_data) - 1;
+		len = MIN(net_buf_simple_pull_u8(adv_data) - 1, sizeof(data) - 1);
```

Comment:
Thanks for catching this! I think it will probably be safer if remove the decrement and do memcpy if `len - 1 > 0`. Otherwise, if `adv_data[0] == 0`, this expression will set `len` to `30` and then it will try to pull 30 bytes from `adv_data` using `net_buf_simple_pull_mem` and it will assert:
https://github.com/zephyrproject-rtos/zephyr/blob/c607179cc46dc260f71e7ba9e6097e0bc3004e92/subsys/net/buf_simple.c#L403-L409

Reply:
> if adv_data[0] == 0, this expression will set len to 30

Right, because of underflow, right? 
So this should be something like 
```c
len = net_buf_simple_pull_u8(adv_data);
type = net_buf_simple_pull_u8(adv_data);
if (len > 0 && len < (sizeof(data) - 1)) {
    memcpy(data, net_buf_simple_pull_mem(adv_data, len), len);
    data[len] = '\0';
} else {
    data[0] =  '\0'; /* Invalid data */
}
```

Or perhaps just ignore and return if the data len is invalid?

Reply:
Yes, ignoring is an option.

Reply:
I change it a little bit and truncate the data to fit in the buffer, my impression is that partial data is still useful to print. But if that does not make sense, just let me know.

The change was :
```
		len = net_buf_simple_pull_u8(adv_data);
		if (len > 0) {
			len = MIN(len, sizeof(data) - 1);
			memcpy(data, net_buf_simple_pull_mem(adv_data, len), len);
		}
		data[len] = '\0';
```

Reply:
`len` should be pulled first. And if it is zero, `type` won't be there:
```suggestion
		len = net_buf_simple_pull_u8(adv_data);
		if (len == 0) {
			/* No data in this AD Structure. */
			continue;
		}

		if (len > adv_data->len) {
			/* Malformed AD Structure. */
			break;
		}

		type = net_buf_simple_pull_u8(adv_data);
		if ((--len) > 0) {
			uint8_t dlen;

			/* Pull all length, but print only what fits into `data` array. */
			dlen = MIN(len, sizeof(data) - 1);
			memcpy(data, net_buf_simple_pull_mem(adv_data, len), dlen);
			len = dlen;
		}
		data[len] = '\0';
```